cli/plugins/test/index.js

```diff
@@ -1,61 +1,86 @@
 'use strict';
 const pa11y = require('pa11y');
+const { Analyzer } = require('hint');
 
 /**
  * Test implements Joy commands for running a11y and static code tests
- * 
+ *
  */
 class Test {
-
   /**
    * @param {*} joy : Instance of a Joy class c/w config.json, env and helper functions
-   * 
+   *
    */
   constructor(joy) {
     const prog = joy.prog;
     prog
       .command('test', 'Run various tests including a11y and static analysis')
-      .argument('<subcommand>', 'subcommand [a11y, static]', ['a11y', 'static'])
+      .argument('<subcommand>', 'subcommand [a11y, staticServer, staticWeb]', ['a11y', 'staticServer', 'staticWeb'])
       .option('-u, --urls <urls>', 'URLs to test', prog.LIST)
       .action(async (args, options, logger) => {
-        logger.debug("arguments: %j", args);
-        logger.debug("options: %j", options);
+        logger.debug('arguments: %j', args);
+        logger.debug('options: %j', options);
         const code = await this[args.subcommand].call(joy, options, logger);
         return code;
       });
   }
 
-
   /**
    * Wraps a11y module
    * Todo: More research to see if this is the best module or test app to use
-   * 
+   *
    * @param {*} options : Additional CLI flags
    * @param {*} logger :  Logger
-   * 
+   *
    */
   async a11y(options, logger) {
     if (this.isJoy()) {
       // Remind the user that if this is a Joy project they may need to serve the app if it is local too
       logger.info('Have you started the local stack?');
     }
 
-    // Promisify and run a11y for each supplied url 
+    // Promisify and run a11y for each supplied url
     const results = await Promise.all(
-      options.urls.map(async url => {
+      options.urls.map(async (url) => {
         return await pa11y(url, options);
       })
     );
 
     // Output the raw result objects to stdout
-    results.forEach(result => {
+    results.forEach((result) => {
       console.log(result);
     });
 
     return 0;
   }
 
+  /**
+   * Wraps [webhint](https://webhint.io/) to provide extensive analysis of front-end code
+   *
+   * @param {*} options : Additional CLI flags
+   * @param {*} logger :  Logger
+   */
+  async staticWeb(options, logger) {
+    const config = {
+      connector: {
+        name: 'jsdom'
+      },
+      extends: ['web-recommended'],
+      formatters: ['stylish'],
+      parsers: ['css', 'html', 'javascript', 'manifest', 'package-json'],
+      browserslist: ['> 1%', 'last 2 versions']
+    };
 
+    const webhint = Analyzer.create(config);
+    const results = await webhint.analyze(options.urls[0]);
+
+    // TODO: Determine how we want to handle output (json to stdout? create excel file? display on screen? something else?)
```

Comment:
I think `stdout` is good for now, perhaps a flag or two down the road to export to a couple basic file types.

Reply:
https://github.com/tforster/joy/issues/11